nacp.go

```diff
@@ -505,16 +505,20 @@ func buildConfig(logger hclog.Logger) *config.Config {
 	return c
 }
 
-func createTlsConfig(caFile string) (*tls.Config, error) {
-	caCert, err := os.ReadFile(caFile)
+func createTlsConfig(c *config.Config) (*tls.Config, error) {
+	caCert, err := os.ReadFile(c.Tls.CaFile)
 	if err != nil {
 		return nil, err
 	}
 	caCertPool := x509.NewCertPool()
 	caCertPool.AppendCertsFromPEM(caCert)
 	tlsConfig := &tls.Config{
-		ClientCAs:  caCertPool,
-		ClientAuth: tls.RequireAndVerifyClientCert,
+		ClientCAs: caCertPool,
+	}
+	if c.Tls.NoClientCert {
```

Comment:
If you don't mind, could you move the clientAuth logic before the struct creation?
e.g:

```go
clientAuth = tls.RequireAndVerifyClientCert
if c.Tls.NoClientCert {
 clientAuth = tls.NoClientCert
}

tlsConfig := &tls.Config{
  ClientCAs:  caCertPool,
  ClientAuth: clientAuth,
  ClientCAs: caCertPool,
}
```

Reply:
yes makes sense, adjusted :)